src/Database/LSMTree/Internal/Snapshot/Codec.hs

```diff
@@ -5,10 +5,16 @@ module Database.LSMTree.Internal.Snapshot.Codec (
   , prettySnapshotVersion
   , currentSnapshotVersion
     -- * Writing and reading files
+    -- ** Metadata
   , writeFileSnapshotMetaData
   , readFileSnapshotMetaData
   , encodeSnapshotMetaData
   , decodeSnapshotMetaData
+    -- ** Merging trees
+  , writeFileSnapshotMergingTree
+  , readFileSnapshotMergingTree
+  , encodeSnapshotMergingTree
+  , decodeSnapshotMergingTree
```

Comment:
I'm a bit confused by this.

I was expecting that `SnapshotMetaData` would be extended to contain the union level / merging tree, and then read/writeFileSnapshotMetaData would be extended to write that out.

I can see it's good to test on its own, but ultimately I think we need it integrated into a single snapshot top level format, which is the `SnapshotMetaData`.

Or is this intended as a placeholder until the top level `snapshot` in `src/Database/LSMTree/Internal.hs` is extended to use the new code?